### DISCUSSION TITLE
Add language navigation to top of README files.

### DISCUSSION THREAD

**Discussion by thias15** [3ru/gpt-translate#56] [Ideas]
### Goals

Prompt the model to update the source README file and README files in other languages with a navigation header, e.g.:
<p align="center">
  <span>English</span> |
  <a href="README_CN.md">简体中文</a> |
  <a href="README.de-DE.md">Deutsch</a>
</p>

### Background

It would be nice to automate adding a multi-language navigation workflow. 

### Proposal

The easiest would probably be to prompt the language model to update a navigation section at the top of each README file. For example, add all existing README files and a prompt like this:
Make sure that {targetLanguage} is linked to the corresponding README file using its Language Culture Name. \
Keep the HTML snippet in the same place as in the source file or add it at the top and keep existing languages. \
For example, for English, Chinese and German the top of the README should look like this:
<p align="center">
  <a href="README.md">English</a> |
  <a href="README.zh-CN.md">简体中文</a> |
  <a href="README.de-DE.md">Deutsch</a>
</p>

**Comment by 3ru** [comment]
Is this a suggestion to include this kind of navigation in the repository? I think it's certainly a good idea.

**Reply by thias15** [reply]
Yes.

**Reply by thias15** [reply]
But more importantly support the automatic generation of such a navigation header as part of creating multi-language docs.

**Comment by thias15** [comment]
What I have done for now is simply use the following prompt. 

```
prompt: |
            Please translate the README source file to {targetLanguage}. \
              Make the translation sound as natural as possible. \
              Keep the HTML snippet with the language options in the same place as in the source file. \
              ONLY {targetLanguage} should be shown in plaintext (between span tags). \
              All other languages should be shown as links using the Language Culture Name, e.g. "README.de-DE.md" for German. \
              Exception: English is the default language and always points to "README.md". \
              For example, the HTML snippet of the German README should look something like this:
              <p align="center">
                <a href="README.md">English</a> |
                <a href="README.zh-CN.md">简体中文</a> |
                <span>Deutsch</span> |
                <a href="README.fr-FR.md">Français</a>
              </p>
              Do NOT add ``` tags to the HTML
```
In order to make sure this works reliably I make sure this HTML header already exists in the original README file. However, it would be even nicer if the action is also able to modify to source README file to update the HTML navigation header, e.g. to add a new language. 

Also, this requires to follow a specific format, e.g. using the language culture name in the extension. It would also be nice for the language model to know the file names / paths for each language for example by making it available as a variable.
